packages/instantsearch.js/src/lib/InstantSearch.ts

```diff
@@ -261,20 +265,30 @@ Use \`InstantSearch.status === "stalled"\` instead.`
       },
     } = options;
 
-    if (searchClient === null) {
-      throw new Error(withUsage('The `searchClient` option is required.'));
+    if (searchClient === null && client === null) {
```

Comment:
```suggestion
    if (searchClient === null || client === null) {
```

Reply:
exactly, that's what I suggest :)

Reply:
You're suggesting that we throw if any of those terms is `false`, but we only need to do it if they both are :)